core/json/mod.rs

```diff
@@ -604,15 +604,21 @@ fn json_path_from_db_value<'a>(
     let path = path.as_value_ref();
     let json_path = if strict {
         match path {
-            ValueRef::Text(t) => json_path(t.as_str())?,
+            ValueRef::Text(t) => match json_path(t.as_str()) {
+                Ok(p) => p,
+                Err(_) => return Ok(None),
+            },
             ValueRef::Null => return Ok(None),
             _ => crate::bail_constraint_error!("JSON path error near: {:?}", path.to_string()),
         }
     } else {
         match path {
             ValueRef::Text(t) => {
                 if t.as_str().starts_with("$") {
-                    json_path(t.as_str())?
+                    match json_path(t.as_str()) {
+                        Ok(p) => p,
+                        Err(_) => return Ok(None),
+                    }
```

Comment:
Should any error return None? I believe there could be type of error while parsing path that shouldn't be allowed or does sqlite treat it differently.